VAMobile/src/screens/HomeScreen/ProfileScreen/SettingsScreen/DeveloperScreen/OverrideApiScreen.tsx

```diff
@@ -0,0 +1,551 @@
+            inputType="phone"
+            onChange={(val) => {
+              const otherErrors = _.remove(overrideErrors, function (n) {
+                return n.endpoint !== endpoint
+              })
+              const backEndErrors = _.remove(overrideErrors, function (n) {
+                return n.endpoint === endpoint
+              })[0]
+
+              const backEndError = backEndErrors.error as errorOverride
+              if (val.length >= 1) {
+                backEndError.status = parseInt(val, 10)
+              } else {
+                backEndError.status = 0
+              }
+
+              otherErrors.push({
+                endpoint,
+                error: backEndError,
+              })
+              setErrors(otherErrors)
+            }}
+          />
+        </Box>
+      )}
+    </Box>
+  )
+}
+
+function OverrideAPIScreen({ navigation }: OverrideAPIScreenProps) {
+  const { t } = useTranslation(NAMESPACE.COMMON)
+  const theme = useTheme()
+  const dispatch = useAppDispatch()
+  const { overrideErrors } = useSelector<RootState, DemoState>((state) => state.demo)
+  const [clearData, setClearData] = useState(false)
+  const [temporaryErrors, setErrors] = useState<Array<errors>>([])
+
+  useEffect(() => {
+    setErrors(overrideErrors)
+  }, [overrideErrors])
+
+  useEffect(() => {
+    if (clearData) {
+      setClearData(false)
+      dispatch(updateErrorOverrides([]))
+    }
+  }, [clearData, dispatch])
+
+  const saveErrors = () => {
+    dispatch(updateErrorOverrides(temporaryErrors))
+  }
+
+  const clearErrors = () => {
+    setClearData(true)
+  }
+
+  return (
+    <FeatureLandingTemplate
+      backLabel={t('debug.title')}
+      backLabelOnPress={navigation.goBack}
+      title={t('overrideAPI')}
+      testID="overrideAPITestID"
+      footerContent={
+        <Box
+          mx={theme.dimensions.gutter}
+          mb={theme.dimensions.contentMarginBottom}
+          mt={theme.dimensions.standardMarginBetween}>
+          <Box mb={theme.dimensions.standardMarginBetween}>
+            <Button label="Set API Errors" onPress={saveErrors} />
+          </Box>
+          <Button label="Clear API Errors" onPress={clearErrors} />
+        </Box>
+      }>
+      <Box>
+        <TextArea>
+          <TextView accessibilityRole="header" variant="MobileBodyBold">
+            Appointments
+          </TextView>
+          <Box my={theme.dimensions.standardMarginBetween}>
+            {IndividualQueryDisplay('/v0/appointments', temporaryErrors, setErrors, clearData, t, theme)}
+          </Box>
+        </TextArea>
+        <TextArea>
+          <TextView accessibilityRole="header" variant="MobileBodyBold">
+            Authorized Services
+          </TextView>
+          <Box my={theme.dimensions.standardMarginBetween}>
+            {IndividualQueryDisplay('/v0/user/authorized-services', temporaryErrors, setErrors, clearData, t, theme)}
+          </Box>
+        </TextArea>
```

Comment:
Instead of repeating the content for each endpoint, couldn't you make an array for the endpoints, and loop over that? e.g.

```tsx
const endpoints = [
  {
    name: 'Appointments',
    endpoint: '/v0/appointments',
  },
  {
    name: 'Authorized Services',
    endpoint: '/v0/user/authorized-services'
  },
  ...
]
```

This would result in less lines of code, and would also make it easier to add endpoints to this list in the future.

Reply:
updated to an array of api groupings